webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteReducer.js

```diff
@@ -0,0 +1,72 @@
+import Immutable from 'seamless-immutable';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+  AUTO_COMPLETE_USED_QUERY,
+  AUTO_COMPLETE_INITIAL_QUERY,
+} from './AutoCompleteConstants';
+
+const initialState = Immutable({
+  controller: null,
+  error: null,
+  results: [],
+  searchQuery: '',
+  status: null,
+  trigger: null,
+  actionsHistory: [],
+});
+
+export default (state = initialState, action) => {
+  const { payload, type } = action;
+  const actionsHistory = state.actionsHistory.set(state.actionsHistory.length, action);
+  const {
+    controller, error, results, searchQuery, status, trigger,
+  } = payload || {};
+  switch (type) {
+    case AUTO_COMPLETE_REQUEST:
+      return state.merge({
+        controller,
+        error: null,
+        searchQuery,
+        status,
+        trigger,
+        actionsHistory,
+      });
+    case AUTO_COMPLETE_SUCCESS:
+      return state.merge({
+        error: null,
+        results,
+        status,
+        actionsHistory,
+      });
+    case AUTO_COMPLETE_FAILURE:
+      return state.merge({
+        error,
+        results: [],
+        status,
+        actionsHistory,
+      });
+    case AUTO_COMPLETE_RESET:
+      return state.merge({
+        ...initialState,
+        actionsHistory,
+      });
+    case AUTO_COMPLETE_USED_QUERY:
+      return state.merge({
+        controller,
+        searchQuery,
+        results,
+        actionsHistory,
+      });
+    case AUTO_COMPLETE_INITIAL_QUERY:
```

Comment:
this still feels like an extra action? does it make sense to combine this event with other events? (such as used_query or success)?

Reply:
We can switch it by the `getResults` action, 
but this means that an API call will be made when the page loads which may cause the loading spinner to be seen, something we wanted to avoid earlier.

the `updateSearchQuery` action, updates the store with the `query` and `controller`, that comes initially from the server side as the component is mounted, without loading the results.

Reply:
Removed the AUTO_COMPLETE_INITIAL_QUERY action,
now using AUTO_COMPLETE_SUCCESS